src/status_emitter.rs

```diff
@@ -228,6 +248,24 @@ impl<T: StatusEmitter> StatusEmitter for (T,) {
     }
 }
 
+impl<T: StatusEmitter + Send> StatusEmitter for (Arc<T>,) {
```

Comment:
```suggestion
impl<T: StatusEmitter + Send> StatusEmitter for Arc<T> {
```

should be the right thing, then the tuple impl will pick it up automatically

Reply:
My colleague actually tried this, and it did not work (I don't have the exact errors at hand). Instead he got this to work (see `patch.txt`).
[patch.txt](https://github.com/user-attachments/files/19889486/patch.txt)

Reply:
you can paste diffs rendered inline:

```diff
diff --git a/src/status_emitter.rs b/src/status_emitter.rs
index f3e8962..0cef7ec 100644
--- a/src/status_emitter.rs
+++ b/src/status_emitter.rs
@@ -6,7 +6,7 @@ use std::{
     fmt::Debug,
     panic::RefUnwindSafe,
     path::{Path, PathBuf},
-    sync::{Arc, Mutex},
+    sync::Arc,
 };
 pub mod debug;
 #[cfg(feature = "gha")]
@@ -36,7 +36,7 @@ mod text;
 // Summary::test_failure().
 
 /// A generic way to handle the output of this crate.
-pub trait StatusEmitter: Sync + RefUnwindSafe {
+pub trait StatusEmitter: Send + Sync + RefUnwindSafe {
     /// Invoked the moment we know a test will later be run.
     /// Useful for progress bars and such.
     fn register_test(&self, path: PathBuf) -> Box<dyn TestStatus + 'static>;
@@ -203,7 +203,7 @@ impl<T: TestStatus, U: TestStatus> TestStatus for (T, U) {
     }
 }
 
-impl<T: StatusEmitter + Send, U: StatusEmitter> StatusEmitter for (Arc<T>, U) {
+impl<T: StatusEmitter, U: StatusEmitter> StatusEmitter for (T, U) {
     fn register_test(&self, path: PathBuf) -> Box<dyn TestStatus> {
         Box::new((
             self.0.register_test(path.clone()),
@@ -248,9 +248,9 @@ impl<T: StatusEmitter> StatusEmitter for (T,) {
     }
 }
 
-impl<T: StatusEmitter + Send> StatusEmitter for (Arc<T>,) {
+impl StatusEmitter for Arc<dyn StatusEmitter> {
     fn register_test(&self, path: PathBuf) -> Box<dyn TestStatus> {
-        self.0.register_test(path.clone())
+        (**self).register_test(path.clone())
     }
 
     fn finalize(
@@ -261,8 +261,7 @@ impl<T: StatusEmitter + Send> StatusEmitter for (Arc<T>,) {
         filtered: usize,
         aborted: bool,
     ) -> Box<dyn Summary> {
-        self.0
-            .finalize(failures, succeeded, ignored, filtered, aborted)
+        (**self).finalize(failures, succeeded, ignored, filtered, aborted)
     }
 }

```

Reply:
> My colleague actually tried this, and it did not work

I copy pasted the `Box<T>` impl to an `Arc<T>` impl and everything still compiled. 

```rust
impl<T: StatusEmitter + ?Sized + Send> StatusEmitter for Arc<T> {
    fn register_test(&self, path: PathBuf) -> Box<dyn TestStatus> {
        (**self).register_test(path)
    }

    fn finalize(
        &self,
        failures: usize,
        succeeded: usize,
        ignored: usize,
        filtered: usize,
        aborted: bool,
    ) -> Box<dyn Summary> {
        (**self).finalize(failures, succeeded, ignored, filtered, aborted)
    }
}
```

Why do you need an `Arc` anyway? clippy and miri were using `Box` just fine so far.

Reply:
Note that up until now, object `text` in
```
    let text = match args.format {
        Format::Terse => status_emitter::Text::quiet(),
        Format::Pretty => status_emitter::Text::verbose(),
    };
```
had type `Text`. Now that we have a new `StatusEmitter`, we need a `dyn StatusEmitter` object. When this object is wrapped in a `Box`, I get the following error:
```
error[E0277]: `dyn status_emitter::StatusEmitter` cannot be sent between threads safely
   --> src/lib.rs:103:9
    |
99  |       run_tests_generic(
    |       ----------------- required by a bound introduced by this call
...
103 | /         (
104 | |             emitter,
105 | |             #[cfg(feature = "gha")]
106 | |             status_emitter::Gha::<true> { name },
107 | |         ),
    | |_________^ the trait `Send` is not implemented for `Unique<dyn status_emitter::StatusEmitter>`
```
This led me to believe (and I am probably wrong) that using `Arc` instead of `Box` will resolve the compilation error.

Reply:
After much fiddling, I think the fix is to add `Send` to the `pub trait StatusEmitter`. Then I can directly use a `Box`.

Reply:
Done.